libopenage/unit/action.cpp

```diff
@@ -790,6 +790,9 @@ GatherAction::GatherAction(Unit *e, UnitReference tar)
 	Unit *target = this->target.get();
 	if (target->has_attribute(attr_type::resource)) {
 		auto &resource_attr = target->get_attribute<attr_type::resource>();
+		if (this->resource_type != resource_attr.resource_type) {
+		  this->entity->get_attribute<attr_type::gatherer>().amount = 0;
```

Comment:
we use tab indentation, you can have a look at `doc/code_style` for more information